### PR TITLE
[Modular] Mesa: Xen adjustments

### PR DESCRIPTION
**Pull request by Stalkeros2**
## About The Pull Request
image
Significantly nerfed the amount of equipment HECU has. It's now storaged inside their vendor.
image
Xen baseturf generator in Xen too, duh.

Fixed 30 round M16 mags' sprites and made them standard.
Fixed baseturf issues in Xen-related areas. Explosions will no longer result in space tiles there.
Gave HECU their own equipment vendor, akin to Assault Operatives' one; in exchange, most of their equipment has been removed, as they start practically naked.
Added MP5: another Mesa-exclusive gun, rapidly shooting 9x19 ammo from a 30 round magazine. Security'd really like to get those.
Removed safeties from Mesa's Glock-17

## How This Contributes To The Skyrat Roleplay Experience
People upstream/here will like occasional Mesa even more, and folks downstream will finally get to test everything out further.

## Changelog

:cl: Stalkeros
qol: Yet another batch of Mesa-related updates.
/:cl:

### CODE REVIEW

**Comment by Stalkeros2** [comment]
OOPS

**Comment by Stalkeros2** [comment]
freaking baseturfs

File: modular_skyrat/modules/black_mesa/code/turfs.dm [hunk 1]
```diff
@@ -13,53 +13,72 @@
...
 /turf/open/water/xen_acid
+	baseturfs = /turf/open/water/xen_acid
 	initial_gas_mix = OPENTURF_DEFAULT_ATMOS
+	planetary_atmos = TRUE
 	color = COLOR_VIBRANT_LIME
 	light_range = 2
 	light_color = COLOR_VIBRANT_LIME
 	/// How much damage we deal if a mob enters us.
-	var/acid_damage = 30
+	var/acid_damage = 10
```
Comment (Gandalf2k15) [inline modular_skyrat/modules/black_mesa/code/turfs.dm]: Keep it at 30

File: modular_skyrat/modules/black_mesa/code/armaments/__armament_bodyarmor.dm [hunk 2]
```diff
@@ -0,0 +1,61 @@
+
+/datum/armament_entry/hecu/bodyarmor
+	category = "Armor Kit"
```
Comment (Zonespace27) [inline modular_skyrat/modules/black_mesa/code/armaments/__armament_bodyarmor.dm:3]: use defines for categories

File: modular_skyrat/modules/black_mesa/code/armaments/_armament_primary.dm [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+/datum/armament_entry/hecu/primary/special
+	subcategory = ARMAMENT_SUBCATEGORY_SPECIAL
+	mags_to_spawn = 2
+
+/datum/armament_entry/hecu/primary/special/sniper_rifle
+	item_type = /obj/item/gun/ballistic/automatic/cfa_rifle
+	max_purchase = 1
+	cost = 15
+
+/obj/item/storage/box/ammo_box/shotgun_12g
```
Comment (Zonespace27) [inline modular_skyrat/modules/black_mesa/code/armaments/_armament_primary.dm:60]: take this to its own file or something

**Comment by Stalkeros2** [comment]
I better continue tomorrow at this point.

**Pull request merged**